### PR TITLE
Feature/norwegian contracts

### PR DESCRIPTION
**Pull request by vonElfvin**
- Wrestle a bit with ExtraBuildingDto
- Include agreementId when creating a contract to be stored in underwriter
- Add tests to `AgreementStatus`

### CODE REVIEW

File: src/main/java/com/hedvig/contract/shared/models/objects/ExtraBuilding.kt [hunk 1]
```diff
@@ -45,6 +45,14 @@ class ExtraBuilding(
     ExtraBuildingType.OTHER -> "Övrigt"
   }
 
+  fun toExtraBuildingDto(localizationService: LocalizationService? = null, locale: Locale? = null) = ExtraBuildingDto(
```
Comment (FredrikAreschoug) [inline src/main/java/com/hedvig/contract/shared/models/objects/ExtraBuilding.kt]: Feels strange to default `localizationService` and `locale` to null we should be more explicit when we don't want to translate

Comment (FredrikAreschoug) [reply src/main/java/com/hedvig/contract/shared/models/objects/ExtraBuilding.kt]: maybe creating a new function called toExtraBuildingEmptyDisplayNameDto when we really don't care about translations

File: src/main/java/com/hedvig/contract/web/dtos/AgreementQuote.kt [hunk 2]
```diff
@@ -18,59 +17,59 @@ import java.util.UUID
...
-  val currency: String
-) {
-  class SwedishApartmentQuote(
-    quoteId: UUID,
-    fromDate: LocalDate?,
-    toDate: LocalDate?,
-    premium: BigDecimal,
-    currency: String,
+sealed class AgreementQuote {
+  abstract val quoteId: UUID
```
Comment (FredrikAreschoug) [inline src/main/java/com/hedvig/contract/web/dtos/AgreementQuote.kt]: :heart_eyes:

File: src/main/java/com/hedvig/contract/shared/models/objects/ExtraBuilding.kt (hunk 1 above)
Comment (vonElfvin) [reply src/main/java/com/hedvig/contract/shared/models/objects/ExtraBuilding.kt]: Split them up now

**Pull request merged**